app/controllers/languages_controller.rb

```diff
@@ -1,5 +1,5 @@
 class LanguagesController < ApplicationController
-  before_filter :check_permission, :only => [:new, :create]
+  before_filter :check_permission, :only => [:new, :create, :edit, :update]
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer `before_action` over `before_filter`.